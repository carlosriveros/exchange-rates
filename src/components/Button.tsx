import React from 'react';
import { Button as AntButton} from 'antd'

interface Props {
    onClick: () => void,
    message: string,
}

const Button = (props : Props) => {
    return <AntButton onClick={props.onClick}>{ props.message }</AntButton>
}

export default Button