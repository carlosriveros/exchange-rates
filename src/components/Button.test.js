import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";
import renderer from "react-test-renderer";

describe("Button", () => {
  const props = {
    message: "I am button",
    onClick: jest.fn()
  };
  it("renders correctly", () => {
    const tree = renderer.create(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Button message", () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper.children().text()).toEqual("I am button");
  });

  it("on click button event calls on click prop", () => {
    const wrapper = shallow(<Button {...props} />);
    wrapper.simulate("click");
    expect(props.onClick).toHaveBeenCalled();
  });
});
