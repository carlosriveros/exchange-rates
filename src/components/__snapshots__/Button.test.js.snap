// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    I am button
  </span>
</button>
`;
